core/main/src/bootstrap/boot.rs

```diff
@@ -46,25 +47,29 @@ use super::{
 ///
 /// # Steps
 ///
-/// 1. [SetupExtnClientStep] - Initializes the extn client to start the Inter process communication backbone
-/// 2. [LoadExtensionMetadataStep] - Loads the Extn metadata from the So files
-/// 3. [LoadExtensionsStep] - Loads the Extensions in to [crate::state::extn_state::ExtnState]
-/// 4. [StartExtnChannelsStep] - Starts the Device channel extension
-/// 5. [StartAppManagerStep] - Starts the App Manager and other supporting services
-/// 6. [LoadDistributorValuesStep] - Loads the values from distributor like Session
-/// 7. [CheckLauncherStep] - Checks the presence of launcher extension and starts default app
-/// 8. [StartWsStep] - Starts the Websocket to accept external and internal connections
-/// 9. [FireboltGatewayStep] - Starts the firebolt gateway and blocks the thread to keep it alive till interruption.
+/// 1. [StartCommunicationBroker] - Initialize the communication broker to create Thunder broker if rules are setup.
```

Comment:
Is there a sequence diagram somewhere that describes the sequence as a request passes through the communication broker, through the thunder broker, endpoint broker, etc.?

Reply:
Yes there is a sequence diagram i will add it to the Ripple as a Gateway documentation[ ](https://wiki.rdkcentral.com/display/RDK/Ripple+as+a+API+Gateway#RippleasaAPIGateway-Brokers)